labs/idp-okta-integration/README.md

```diff
@@ -0,0 +1,302 @@
+
+     ![Okta Auth](assets/okta-auth-code-login.png)
+
+3. After successful authentication, Okta redirects to the Apigee callback URL (/v1/oauth20/callback), which controls the incoming query parameters, generate an authorization code (using the same value as the one provided by Okta) and performs a redirection on the client app redirect_uri `https://httpbin.org/get` providing the authorization_code and initial state parameters.
+
+     - In a real-world scenario, the redirection would be back to the client application and it would parse Okta's response to capture the authorization code
+
+     ![Okta auth code](assets/okta-auth-code-response.png)
+
+4. Pass the authorization code, client id, and base64 encoded string to Apigee Identity Facade to generate a Bearer token that will be used for API calls.
+
+     ```bash
+     export AUTH_CODE={authorization code returned above}
+     export APIGEE_RESPONSE=$(curl -s --location --request POST "https://$APIGEE_X_HOSTNAME/v1/oauth20/token?client_id=$APIGEE_CLIENT_ID" \
+     --header "Authorization: Basic $BASE64_ENCODED" \
+     --header 'Content-Type: application/x-www-form-urlencoded' \
+     --data-urlencode 'redirect_uri=https://httpbin.org/get' \
+     --data-urlencode 'grant_type=authorization_code' \
+     --data-urlencode "code=$AUTH_CODE")
+     echo $APIGEE_RESPONSE
+     export ACCESS_TOKEN=$(echo $APIGEE_RESPONSE | jq -r .access_token)
+     echo $ACCESS_TOKEN
+     ```
+
+5. Apigee will respond with a Bearer token (access_token) which is saved to an environment variable
+
+6. Finally, call an endpoint protected by OAuth. The identity facade has /protected one configured to validate OAuth tokens.
+
+     ```bash
+     curl --location --request GET "https://$APIGEE_X_HOSTNAME/v1/oauth20/protected" \
+     --header "Authorization: Bearer $ACCESS_TOKEN"
+     ```
+
+     ![IDP Test](assets/idp-test-proxy.png)
+
+## Use Identity Facade with API Proxy
+
+The default Apigee install includes a /hello-world proxy. In this section we will add an OAuthv2 policy that verifies the access token and create an application that has access to the identity facade (generate tokens) and the Hellow-World proxies.
+
+1. In the Apigee portal, expand Publish and click on API Products.
+2. Choose + CREATE
+3. Fill in the following
+
+     - Name: Hellow World
+     - Display Name: Hello World
+     - Environment: eval
+     - Access: Public
+     - Operations (ADD AN OPERATION)
+     - API Proxy: hello-world
+     - Path: /
+     - Save
+     - Scroll up and click Save
+
+     ![API Product1](assets/api-product1.png)
+     ![API Product2](assets/api-product2.png)
+
+4. Under Public, click on Apps.
+
+     - Notice there is an app created called identity app. The client id and secret used above are from this application.
+
+5. Click on +App and configure it as follows:
+
+     - Name: Hello World App
+     - Developer: Jane Doe (this user was also created during the identity facade deployment)
+     - Productt:
```

Comment:
Product

Reply:
@Teodelas please see the trivial typo above. After that we're good to merge. Thanks in advance!